src/FilterFormatManager.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_migration;
+
+use Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException;
+use Drupal\Component\Plugin\Exception\PluginNotFoundException;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\filter\Entity\FilterFormat;
+use Drupal\filter\FilterPluginCollection;
+
+/**
+ * Class FilterFormatManager.
+ *
+ * Provides filter format related methods.
+ *
+ * @package Drupal\oe_migration
+ */
+class FilterFormatManager {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The filter format storage.
+   *
+   * @var \Drupal\Core\Config\Entity\ConfigEntityStorage
+   */
+  protected $filterFormatStorage;
+
+  /**
+   * The FilterFormatManager constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->setFilterFormatStorage();
+  }
+
+  /**
+   * The filterFormatStorage setter.
+   */
+  public function setFilterFormatStorage(): void {
+    try {
+      $this->filterFormatStorage = $this->entityTypeManager->getStorage('filter_format');
+    }
+    catch (PluginNotFoundException | InvalidPluginDefinitionException $e) {
+      $this->filterFormatStorage = NULL;
+    }
+  }
+
+  /**
+   * Returns a filter format by its id.
+   *
+   * @param string $id
+   *   The filter format id.
+   *
+   * @return \Drupal\filter\Entity\FilterFormat|null
+   *   The filter format object or null.
+   */
+  public function getFilterFormat(string $id): ?FilterFormat {
+    return $this->filterFormatStorage ? $this->filterFormatStorage->load($id) : NULL;
+  }
+
+  /**
+   * Returns a list of filter id's for a given filter format.
+   *
+   * @param \Drupal\filter\Entity\FilterFormat $filter_format
+   *   The filter format object.
+   *
+   * @return array
+   *   The list of filter id's for a given filter format or an empty list.
+   */
+  public function getFilterIds(FilterFormat $filter_format): array {
```

Comment:
Always typehint (both in PHP and in comments using interfaces and not the actual classes). See everywhere please.

Reply:
**Status:** Fixed
Typehints have been replaced. Interfaces are used instead of classes.